### PR TITLE
added single article components

### PR DESCRIPTION
**Pull request by MolchanovaA**

### CODE REVIEW

File: src/components/SingleArticle.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { getArticleById } from "../utils/apiRequests";
+
+import "./styles/singleArticle.css";
+
+const SingleArticle = () => {
+  const [article, setArticle] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [date, setDate] = useState("");
```
Comment (ghost) [inline src/components/SingleArticle.jsx]: Does this really need state??

File: src/components/SingleArticle.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  const params = useParams();
+  const { article_id } = params;
+
+  useEffect(() => {
+    getArticleById(article_id).then((data) => {
+      setArticle(data);
+      const dateCreated = data.created_at.split("T");
+      setDate(dateCreated[0]);
```
Comment (ghost) [inline src/components/SingleArticle.jsx]: There is a better way of handling dates, take a look at day.js or update the object before setting it in state as you have access to it.

File: src/components/routes/RutesComponent.jsx [hunk 3]
```diff
@@ -3,15 +3,19 @@ import { Route, Routes } from "react-router-dom";
...
-    <Routes>
-      <Route path="/" element={<WebTemplate />}>
-        <Route path="api" element={<TemplateBody />} />
-        <Route path="api/articles" element={<Articles />} />
-      </Route>
-    </Routes>
+    <>
+      <Routes>
+        <Route path="/" element={<WebTemplate />}>
+          <Route path="api" element={<TemplateBody />} />
```
Comment (ghost) [inline src/components/routes/RutesComponent.jsx]: What is this route doing? Why would there be a path to api?

**Review by ghost — APPROVED** [review]
Functionally this is great,  a single article is rendered very nicely. I'm not sure why your url is including /api/ however.

You don't need to handle your dates in state, you can absolutely handle the date before you hand it to the article state.

File: src/components/routes/RutesComponent.jsx [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
-      <Route path="/" element={<WebTemplate />}>
-        <Route path="api" element={<TemplateBody />} />
+    <>
+      <Routes>
+        <Route path="/" element={<Home />} />
         <Route path="api/articles" element={<Articles />} />
-      </Route>
-    </Routes>
+        <Route path="api/articles/:article_id" element={<SingleArticle />} />
+      </Routes>
```
Comment (popatre) [inline src/components/routes/RutesComponent.jsx]: These routes still aren't quite right. You url has /api in it. You're not building an api, you're building a website so the urls don;t really make sense

**Review by popatre — CHANGES_REQUESTED** [review]
These routes still aren't quite right. You url has /api in it. You're not building an api, you're building a website so the urls don;t really make sense

**Comment by ghost** [comment]
Nicely done, don't forget to put a loading pattern in on this. You don't need to resubmit that though :) just on a slow connection it isn't clear that its working.

**Pull request merged**